test/cli_test.py

```diff
@@ -350,6 +371,175 @@ def test_unkown_collection(self):
+            }
+        }
+        self.external_config = self.config["alternatives"]["myexternal"]
+
+    def touch_art(self, artpath: bytes, image_path: Path):
+        """`touch` the image file, but don't set mtime to the current
+        time since the tests run rather fast and item and art mtimes might
+        end up identical if the filesystem has low mtime granularity or
+        mtimes are cashed as laid out in
+            https://stackoverflow.com/a/14393315/3451198
+        Considering the interpreter startup time when running `beet alt
+        update <name>` in a real use-case, this should not obscure any
+        bugs.
+        """
+        item_mtime_alt = Path(str(artpath, "utf8")).stat().st_mtime
+        os.utime(image_path, (item_mtime_alt + 2, item_mtime_alt + 2))
+
+    def test_resize_art(self, tmp_path: Path):
+        album = self.add_album(myexternal="true")
+        album.store()
+        self.runcli("alt", "update", "myexternal")
+
+        # Make a copy of the artwork, so that changing mtime/content won't
+        # affect the repository.
+        image_path = tmp_path / "FIXTURE.png"
+        artpath = bytes(image_path)
+        shutil.copy(self.IMAGE_FIXTURE1, image_path)
+        self.touch_art(artpath, image_path)
+
+        dest_dir = self.get_album_path(album)
+        dest = album.art_destination(artpath, bytes(dest_dir, encoding="utf8"))
+
+        self.external_config["album_art_copy"] = True
+        self.external_config["album_art_maxwidth"] = 1
+        album.artpath = artpath
+        album.store()
+        self.runcli("alt", "update", "myexternal")
+        (width, height) = ArtResizer.shared.get_size(path_in=dest)
+        assert width == 1
+        assert height < 3
+        assert ArtResizer.shared.get_format(path_in=dest) == "PNG"
+
+        self.external_config["album_art_format"] = "JPEG"
+        self.runcli("alt", "update", "myexternal")
+
+        dest = album.art_destination(
+            tmp_path / "FIXTURE.jpg", bytes(dest_dir, encoding="utf8")
+        )
+
+        (width, height) = ArtResizer.shared.get_size(path_in=dest)
+        assert width == 1
+        assert height < 3
+        assert ArtResizer.shared.get_format(path_in=dest) == "JPEG"
+        assert Path(str(dest, "utf8")).name == "cover.jpg"
+        # Check that FIXTURE.png is still around to verify that
+        # the reformat was not done in-place
+        assert image_path.is_file()
+
+        # Test that reformat is idempotent
+        image_path = tmp_path / "FIXTURE.jpg"
+        util.copy(dest, image_path)
+        album.artpath = bytes(image_path)
+        album.store()
+        self.runcli("alt", "update", "myexternal")
+
+        dest_path = Path(str(dest, "utf8"))
+        mtime_1 = dest_path.stat().st_mtime
+        self.runcli("alt", "update", "myexternal")
+        mtime_2 = dest_path.stat().st_mtime
+
+        assert mtime_1 == mtime_2
+
+    def test_copy_art(self, tmp_path: Path):
```

Comment:
It would be great if we had a test case that checks that the art is only reprocessed and copied if the timestamp is newer than the existing art timestamp.